activation/activation.go

```diff
@@ -594,15 +605,8 @@ func (b *Builder) PublishActivationTx(ctx context.Context) error {
 	case <-atxReceived:
 		logger.With().Info("received atx in db", atx.ID())
 	case <-b.layerClock.AwaitLayer((atx.TargetEpoch() + 1).FirstLayer()):
-		select {
-		case <-atxReceived:
-			logger.With().Info("received atx in db (in the last moment)", atx.ID())
```

Comment:
why is this removed? select can pick random case right?

Reply:
`select` will pick a random out of all ready cases. I am not sure if this is a problem. It depends if "receiving" ATX happening AFTER the target epoch is still OK. I think it's basically useless at this point, isn't it?

Reply:
`select` will pick the case that is satisfied. If more than one case is satisfied simultaneously it will pock a random case. Here the check was the following:

If the layer was reached it would check again if the ATX was received. This again has a chance to not be picked since `RegisterForATXSync` would always be satisfied. So if the ATX was received right when the layer switched it might still not be handled but instead marked as `ErrATXChallengeExpired`.

Essentially for this case to be selected the following has to happen:

1. ATX must be received within a very short time period (few ms) before `TargetEpoch()` ends
2. The `AwaitLayer` signal must be picked over the `atxReceived` by the `select` since both are satisfied (50:50 chance)
3. Then the next select must select `atxReceived` when it also could always select on `RegisterATXSynced` (another 50:50 chance)

This whole code block is there to detect an ATX that is "in the last moment" with 25% chance. In the last 2 days not one system test contained it: [Grafana](https://grafana.spacemesh.dev/explore?orgId=1&left=%7B%22datasource%22:%22loki-gke-dev%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-gke-dev%22%7D,%22editorMode%22:%22builder%22,%22expr%22:%22%7Bcontainer%3D%5C%22smesher%5C%22%7D%20%7C%3D%20%60in%20the%20last%20moment%60%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%22now-24h%22,%22to%22:%22now%22%7D%7D)